### DISCUSSION TITLE
Bot ignoring my stores

### DISCUSSION THREAD

**Discussion by Mr-Tech-13** [jef/streetmerchant#2275] [Help] [answered]
Well, I'm adding stores to street merchant, and every time, I save and street merchant ignores it. Same goes for everything.
<img width="244" alt="Screen Shot 2021-04-02 at 12 13 50 PM" src="https://user-images.githubusercontent.com/81822275/113433346-09ebb680-93ad-11eb-819c-87adec8a40d7.png">
<img width="555" alt="Screen Shot 2021-04-02 at 12 14 27 PM" src="https://user-images.githubusercontent.com/81822275/113433347-0a844d00-93ad-11eb-9ff6-117476b6ead0.png">

**Comment by aradenac** [comment, marked as answer]
Maybe a basic question, but did your copy and rename streetmerchant/dotenv-example to streetmerchant/dotenv ?

**Reply by Mr-Tech-13** [reply]
Yes

**Reply by aradenac** [reply]
Ok, I asked this because despite the SHOW_ONLY_SERIES=3060,3060ti,3070 of your dotenv file, you get all the series defined by default in src/config.ts:

> 420   showOnlySeries: envOrArray(process.env.SHOW_ONLY_SERIES, [
421     '3060',
422     '3060ti',
423     '3070',
424     '3080',
425     '3090',
426     'rx6800',
427     'rx6800xt',
428     'rx6900xt',
429     'ryzen5600',
430     'ryzen5800',
431     'ryzen5900',
432     'ryzen5950',
433     'sf',
434     'sonyps5c',
435     'sonyps5de',
436     'xboxss',
437     'xboxsx',
438   ]),

That means for whatever reason your dotenv seems to be ignored.

Concerning the stores that you want to add, ensure you added them to STORES in your dotenv. By example in my case I have:
112 STORES=materiel.net, ldlc, amazon-fr

'materiel.net' being the new store I created.

**Reply by aradenac** [reply]
Can you see these logs when you run "npm start run":
> streetmerchant@3.5.0 prestart
> npm run compile

> streetmerchant@3.5.0 compile
> tsc

> streetmerchant@3.5.0 start
> node build/src/index.js

As the dotenv configuration needs to be recompiled after modification ?

**Reply by Mr-Tech-13** [reply]
> streetmerchant@3.5.0 prestart /Users/maxxshearey/Downloads/streetmerchant
> npm run compile


> streetmerchant@3.5.0 compile
> tsc


> streetmerchant@3.5.0 start /Users/maxxshearey/Downloads/streetmerchant
> node build/src/index.js "run"

This is what I see, before modification

**Reply by Mr-Tech-13** [reply]
ok, just deleted what I dine want, and added stores, and presto it works!! Thank you so much for your time

**Reply by Mr-Tech-13** [reply]
How do I add a discord notification if I can't use dotnev?

**Reply by aradenac** [reply]
Fine ! If I get it right, you edited src/config.ts to delete what you do not need and add what is relevant for you in the default values ?

If that is the case, the discord parameters are located in src/config.ts starting at line 216, you can add a default value to the parameters as by now an empty string array is generated by envOrArray() when no default is provided.

By example, in line 217 you can add a second argument to envOrArray, thus replacing:
`notifyGroup: envOrArray(process.env.DISCORD_NOTIFY_GROUP)`
by
`notifyGroup: envOrArray(process.env.DISCORD_NOTIFY_GROUP, ['your_group_id'])`

Same goes for other relevant parameters.

hope that helps, but I still wonder why you cannot use dotenv

**Reply by Mr-Tech-13** [reply]
Figured it out! Dotenv was called dotnev in my folder LOL

**Reply by Mr-Tech-13** [reply]
So how do I setup notifications regularly using dotenv?

**Reply by aradenac** [reply]
Haha classic, I do the same very often ^^, for my part I did not use discord notification so I can't help you for this. But for the gmail notifications you just have to set EMAIL_PASSWORD, EMAIL_TO, EMAIL_USERNAME with your credentials (for me email to and username are the same). 

Then you can run (important):
`npm run test:notification` to validate your dotenv config. You can do it for any notification medium you configured. Google may send you push notifications for security to allow the bot to send mail via your gmail account.

And that's it. I also did configure SMS notifications but as I reside in France I did add a new carrier. If you want to take a look [the pull request is here](https://github.com/jef/streetmerchant/pull/2285)

Good luck with your bot ;)

**Reply by Mr-Tech-13** [reply]
Can you notify a certain role? like @3060?

**Reply by Mr-Tech-13** [reply]
Sussed that out as well. Thank you again for your time

**Comment by 0000day** [comment]
Be aware that if you choose stores which don't have your selected series/brands/models, the bot will just do _nothing_. There have to be product links for what ever you are looking for.

I've created a PR for this topic: https://github.com/jef/streetmerchant/pull/2312
